### DISCUSSION TITLE
use Dogecoin node in project

### DISCUSSION THREAD

**Discussion by yahya-aghdam** [dogecoin/dogecoin#2514] [Q&A] [answered]
Is there a library to match it with Dogecoin node to create transactions and from the Backend?

Thanks 🥇

**Comment by Star6654** [comment]
Ok

**Comment by AbcSxyZ** [comment, marked as answer]
There is no official library to create transaction, the only supported tools to do it are `dogecoin-cli` and `dogecoin-tx` executables.

## JSON-RPC

As basis, the `dogecoind` expose a JSON-RPC interface to query the blockchain and interact with your wallet. At least, you can do http requests directly using any language.
Dogecoin repo do not provide documentation about the RPC API, but you can take a look at the [bitcoin API](https://developer.bitcoin.org/reference/rpc/) for now to build your tools. Almost all commands are the same.

You can find a tutorial to do it with the following [bitcoin resource](https://en.bitcoinwiki.org/wiki/JSON-RPC).

## Non-official Libraries for RPC interface

As non-official libraries, who may or may not work, may or may not be well documented, I found :
 - **Python** : [bmwant/python-dogecoin](https://github.com/bmwant/python-dogecoin)
 - **Javascript** : [Amirh24/bitcoinjs-dogecoin-lib](https://github.com/Amirh24/bitcoinjs-dogecoin-lib) or [dogechaininfo/dogecoinjs](https://github.com/dogechaininfo/dogecoinjs)
 - **Java** : [dogecoin/libdohj](https://github.com/dogecoin/libdohj)
 - ... *If someone share any other libs, I can add them*
 
 ## Rosetta standard
 
 There is also a standard called [Rosetta](https://www.rosetta-api.org/), which is an *"open standard designed to simplify blockchain deployment and **interaction**. Spend less time on integration and more time on novel blockchain advancements."*.
 
 The standard was originally created by Coinbase to interact with [bitcoin](https://github.com/coinbase/rosetta-bitcoin), but it propagated to other cryptos and the implementation is also available for [Dogecoin](https://github.com/rosetta-dogecoin/rosetta-dogecoin).
 
It may be interesting if you want to interact with multiple cryptos who use it, but for now I personally don't know how things works.

Good luck !
